src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar.java

```diff
@@ -98,7 +98,7 @@ public InheritManifest getManifest() {
     @Override
     protected CopyAction createCopyAction() {
         DocumentationRegistry documentationRegistry = getServices().get(DocumentationRegistry.class);
-        final UnusedTracker unusedTracker = minimizeJar ? UnusedTracker.forProject(getProject(), configurations, dependencyFilterForMinimize) : null;
+        final UnusedTracker unusedTracker = minimizeJar ? UnusedTrackerWithR8.forProject(getProject(), configurations, dependencyFilterForMinimize) : null;
```

Comment:
That choice of using the standard tracker vs the r8 tracker needs to be configurable by the user. 
A couple of options here:
1. Add an `useR8 = true` option that works in conjunction with `minimizeJar` to configure the implementation.
2. Detect the presence of the R8 classes on the classpath and then use the R8 implementation if present.